File: internal/cli/status.go
package cli

import (
	"fmt"

	"github.com/delightfulhammers/telesis/internal/status"
	"github.com/spf13/cobra"
)

func init() {
	rootCmd.AddCommand(statusCmd)
}

var statusCmd = &cobra.Command{
	Use:   "status",
	Short: "Print current project state",
	Long:  "Displays the project name, status, ADR/TDD counts, active milestone, and last context generation time.",
	RunE:  runStatus,
}

func runStatus(cmd *cobra.Command, args []string) error {
	rootDir, err := projectRoot()
	if err != nil {
		return fmt.Errorf("could not determine project root: %w", err)
	}

	s, err := status.GetStatus(rootDir)
	if err != nil {
		return err
	}

	fmt.Printf("Project:    %s\n", s.ProjectName)
	fmt.Printf("Status:     %s\n", s.ProjectStatus)
	fmt.Printf("ADRs:       %d\n", s.ADRCount)
	fmt.Printf("TDDs:       %d\n", s.TDDCount)

	if s.ActiveMilestone != "" {
		fmt.Printf("Milestone:  %s\n", firstLine(s.ActiveMilestone))
	} else {
		fmt.Println("Milestone:  (none)")
	}

	if !s.ContextGeneratedAt.IsZero() {
		fmt.Printf("CLAUDE.md:  last generated %s\n", s.ContextGeneratedAt.Format("2006-01-02 15:04:05"))
	} else {
		fmt.Println("CLAUDE.md:  not yet generated")
	}

	return nil
}

func firstLine(s string) string {
	for i, c := range s {
		if c == '\n' {
			return s[:i]
		}
	}
	return s
}

File: src/cli/status.ts
import { Command } from "commander";
import { getStatus } from "../status/status.js";
import { projectRoot } from "./project-root.js";
import { handleAction } from "./handle-action.js";

const firstLine = (s: string): string => {
  const idx = s.indexOf("\n");
  return idx >= 0 ? s.slice(0, idx) : s;
};

const formatNumber = (n: number): string => n.toLocaleString("en-US");

const formatDate = (d: Date): string => {
  const pad = (n: number): string => String(n).padStart(2, "0");
  return `${d.getFullYear()}-${pad(d.getMonth() + 1)}-${pad(d.getDate())} ${pad(d.getHours())}:${pad(d.getMinutes())}:${pad(d.getSeconds())}`;
};

export const statusCommand = new Command("status")
  .description("Print current project state")
  .action(
    handleAction(() => {
      const rootDir = projectRoot();
      const s = getStatus(rootDir);

      console.log(`Project:    ${s.projectName}`);
      console.log(`Status:     ${s.projectStatus}`);
      console.log(`ADRs:       ${s.adrCount}`);
      console.log(`TDDs:       ${s.tddCount}`);

      if (s.activeMilestone) {
        console.log(`Milestone:  ${firstLine(s.activeMilestone)}`);
      } else {
        console.log("Milestone:  (none)");
      }

      if (s.contextGeneratedAt) {
        console.log(
          `CLAUDE.md:  last generated ${formatDate(s.contextGeneratedAt)}`,
        );
      } else {
        console.log("CLAUDE.md:  not yet generated");
      }

      if (s.modelCallCount > 0) {
        console.log(
          `Tokens:     ${formatNumber(s.totalInputTokens)} in / ${formatNumber(s.totalOutputTokens)} out (${s.modelCallCount} call${s.modelCallCount === 1 ? "" : "s"})`,
        );
        if (s.estimatedCost !== null) {
          console.log(`Est. cost:  $${s.estimatedCost.toFixed(4)}`);
        }
      }
    }),
  );
